packages/components/src/components/domainproperties/samples/SampleTypeDesigner.tsx

```diff
@@ -0,0 +1,394 @@
+
+
+        let aliases = {};
+
+        if (parentAliases) {
+            parentAliases.map((alias: IParentAlias) => {
+                const {parentValue} = alias;
+                let value = parentValue.value as string;
+                if (parentValue === NEW_SAMPLE_SET_OPTION)
+                    value = SAMPLE_SET_IMPORT_PREFIX + name;
+
+                aliases[alias.alias] = value;
+            });
+        }
+
+        return Map<string,string>(aliases);
+    };
+
+    onFieldChange = (newModel: SampleTypeModel) => {
+        this.setState(()=>({model:newModel}));
+    };
+
+    parentAliasChange = (id:string, field: string, newValue: any) => {
+        const {model} = this.state;
+        const {parentAliases} = model;
+        const changedAlias = {
+            ...parentAliases.get(id),
+            [field]: newValue,
+        };
+
+        const newAliases = parentAliases.set(id, changedAlias);
+        const newModel = model.merge({parentAliases: newAliases}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    addParentAlias = (id:string, newAlias: IParentAlias): void => {
+        const {model} = this.state;
+        let {parentAliases} = model;
+        const newModel = model.merge({parentAliases:parentAliases.set(id, newAlias)}) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    removeParentAlias = (id:string) => {
+        const {model} = this.state;
+        let {parentAliases} = model;
+        const aliases = parentAliases.delete(id);
+        const newModel = model.set('parentAliases', aliases) as SampleTypeModel;
+        this.setState(() => ({model: newModel}));
+    };
+
+    renderDetailsPanel = () => {
+        const {noun = 'Sample Type', nameExpressionInfoUrl, nameExpressionPlaceholder } = this.props;
+        const {model, parentOptions,} = this.state;
+        return (
+            <>
+                <SampleTypePropertiesPanel
+                    model={model}
+                    parentOptions={parentOptions}
+                    onParentAliasChange={this.parentAliasChange}
+                    onAddParentAlias={this.addParentAlias}
+                    onRemoveParentAlias={this.removeParentAlias}
+                    noun={noun}
+                    nameExpressionInfoUrl={nameExpressionInfoUrl}
+                    nameExpressionPlaceholder={nameExpressionPlaceholder}
+                    updateModel={this.onFieldChange}
+                />
+            </>
+        )
+    };
+
+    domainChangeHandler = (newDomain: DomainDesign): void => {
+        const {defaultSampleFieldConfig} = this.props;
+        let {model} = this.state;
+        let invalidDomainField = undefined;
+        if (newDomain && newDomain.fields) {
+            newDomain.fields.forEach(field => {
+                if (field && field.name && field.name.toLowerCase() === defaultSampleFieldConfig.name.toLowerCase()) {
+                    invalidDomainField = field.name
```

Comment:
I'm not understanding the invalidDomainField variable here. So if I am reading this correctly if the name of a file in the domain matches the defaultSampleFieldConfig then it is invalid? Oh is this a way to validate that the user doesn't use the reserved field name of SampleId or something like that? I'm not seeing where the invalidDomainField state variable is used in the render method of this component.